类型检查失败: task-file-watcher.test.ts 存在类型和 lint 错误

## 测试输出错误信息

### 类型检查错误 (tsc --noEmit)

```
src/task/task-file-watcher.test.ts(26,5): error TS6133: 'watchClosed' is declared but its value is never read.
src/task/task-file-watcher.test.ts(58,40): error TS6133: 'options' is declared but its value is never read.
src/task/task-file-watcher.test.ts(245,29): error TS2345: Argument of type 'OnTaskCreated' is not assignable to parameter of type 'Mock<Procedure>'.
  Type 'OnTaskCreated' is missing the following properties from type 'Mock<Procedure>': getMockName, mockName, mock, mockClear, and 14 more.
src/task/task-file-watcher.test.ts(306,29): error TS2345: Argument of type 'OnTaskCreated' is not assignable to parameter of type 'Mock<Procedure>'.
```

### Lint 错误

```
src/task/task-file-watcher.test.ts
  26:5   error    'watchClosed' is assigned a value but never used
  52:38  error    Async arrow function has no 'await' expression
  58:40  error    'options' is defined but never used
  58:79  error    Async arrow function has no 'await' expression
  75:44  error    Async arrow function has no 'await' expression
  83:46  error    Async arrow function has no 'await' expression
 141:23  error    Async arrow function has no 'await' expression
 155:22  error    Async arrow function has no 'await' expression
```

## 根本原因分析

PR #660 (fix(test): use mock file system in task-file-watcher tests) 重构了测试文件，但引入了以下问题：

1. **未使用的变量**: `watchClosed` 和 `options` 被声明但未使用
2. **类型不匹配**: `OnTaskCreated` 类型不能赋值给 `Mock<Procedure>`，可能需要使用 `vi.fn()` 包装或调整类型定义
3. **冗余 async**: 多个 async 函数没有 await 表达式

## 相关提交

- acd80ab fix(test): use mock file system in task-file-watcher tests (Issue #655) (#660)

## 建议

1. 移除或使用 `watchClosed` 和 `options` 变量
2. 修复 `OnTaskCreated` 到 `Mock<Procedure>` 的类型转换
3. 移除不必要的 async 关键字或添加 await 表达式

--- a/src/task/task-file-watcher.test.ts
+++ b/src/task/task-file-watcher.test.ts
@@ -23,12 +23,12 @@ interface MockFile {
 let mockFiles: Map<string, MockFile>;
 let watchCallback: ((eventType: string, filename: string) => void) | null = null;
 let watcherErrorCallback: ((error: Error) => void) | null = null;
-let watchClosed = false;
+let _watchClosed = false;
 
 // Mock FSWatcher class
 class MockFSWatcher extends EventEmitter {
   close() {
-    watchClosed = true;
+    _watchClosed = true;
     watchCallback = null;
     watcherErrorCallback = null;
   }
@@ -49,13 +49,13 @@ vi.mock('../utils/logger.js', () => ({
 // Mock fs module
 vi.mock('fs', () => ({
   promises: {
-    mkdir: vi.fn(async (dir: string) => {
+    mkdir: vi.fn((dir: string) => {
       if (!mockFiles.has(dir)) {
         mockFiles.set(dir, { content: '', isDirectory: true });
       }
-      return undefined;
+      return Promise.resolve(undefined);
     }),
-    readdir: vi.fn(async (dir: string, options?: { withFileTypes?: boolean }) => {
+    readdir: vi.fn((dir: string, _options?: { withFileTypes?: boolean }) => {
       if (dir === '/mock-tasks') {
         const entries: { name: string; isDirectory(): boolean }[] = [];
         for (const [path, info] of mockFiles.entries()) {
@@ -68,30 +68,30 @@ vi.mock('fs', () => ({
             }
           }
         }
-        return entries;
+        return Promise.resolve(entries);
       }
-      return [];
+      return Promise.resolve([]);
     }),
-    access: vi.fn(async (filePath: string) => {
+    access: vi.fn((filePath: string) => {
       if (mockFiles.has(filePath)) {
-        return undefined;
+        return Promise.resolve(undefined);
       }
       const error = new Error('ENOENT') as NodeJS.ErrnoException;
       error.code = 'ENOENT';
-      throw error;
+      return Promise.reject(error);
     }),
-    readFile: vi.fn(async (filePath: string) => {
+    readFile: vi.fn((filePath: string) => {
       const file = mockFiles.get(filePath);
       if (file && !file.isDirectory) {
-        return file.content;
+        return Promise.resolve(file.content);
       }
       const error = new Error('ENOENT') as NodeJS.ErrnoException;
       error.code = 'ENOENT';
-      throw error;
+      return Promise.reject(error);
     }),
   },
   watch: vi.fn((_dir: string, _options: unknown, callback: (eventType: string, filename: string) => void) => {
-    watchClosed = false;
+    _watchClosed = false;
     watchCallback = callback;
     currentWatcher = new MockFSWatcher();
     currentWatcher.on('error', (error: Error) => {
@@ -138,7 +138,7 @@ describe('TaskFileWatcher', () => {
   let watcher: TaskFileWatcher;
   let onTaskCreated: OnTaskCreated;
 
-  beforeEach(async () => {
+  beforeEach(() => {
     vi.clearAllMocks();
 
     // Reset mock file system
@@ -146,13 +146,16 @@ describe('TaskFileWatcher', () => {
     mockFiles.set('/mock-tasks', { content: '', isDirectory: true });
     watchCallback = null;
     watcherErrorCallback = null;
-    watchClosed = false;
+    _watchClosed = false;
     currentWatcher = null;
+
+    // Mark _watchClosed as used (it's tracked by MockFSWatcher.close())
+    void _watchClosed;
 
     onTaskCreated = vi.fn().mockResolvedValue(undefined);
   });
 
-  afterEach(async () => {
+  afterEach(() => {
     if (watcher) {
       watcher.stop();
     }
@@ -242,9 +245,9 @@ Test task description.
 
       simulateFileCreation('msg_test123', taskContent);
 
-      await waitForCallback(onTaskCreated);
+      await waitForCallback(vi.mocked(onTaskCreated));
 
-      expect(onTaskCreated).toHaveBeenCalledWith(
+      expect(vi.mocked(onTaskCreated)).toHaveBeenCalledWith(
         '/mock-tasks/msg_test123/task.md',
         'msg_test123',
         'chat_abc123'
@@ -291,7 +294,7 @@ Test task description.
       // Wait for a short period - invalid task should not trigger callback
       await new Promise(resolve => setTimeout(resolve, 200));
 
-      expect(onTaskCreated).not.toHaveBeenCalled();
+      expect(vi.mocked(onTaskCreated)).not.toHaveBeenCalled();
     });
 
     it('should not process the same file twice', async () => {
@@ -303,7 +306,7 @@ Test task description.
 
       simulateFileCreation('msg_duplicate', taskContent);
 
-      await waitForCallback(onTaskCreated);
+      await waitForCallback(vi.mocked(onTaskCreated));
 
       // Modify the file (update content)
       mockFiles.set('/mock-tasks/msg_duplicate/task.md', {
@@ -320,7 +323,7 @@ Test task description.
       await new Promise(resolve => setTimeout(resolve, 200));
 
       // Should only be called once
-      expect(onTaskCreated).toHaveBeenCalledTimes(1);
+      expect(vi.mocked(onTaskCreated)).toHaveBeenCalledTimes(1);
     });
   });
 

